dao/elasticsearch/controlplane.json

```diff
@@ -2,36 +2,27 @@
   "settings": {
     "number_of_shards": 1
   },
-
   "mappings": {
-    "servicestate": {
-      "properties": {
-        "Id" :             {"type": "string", "index":"not_analyzed"},
-        "HostID":          {"type": "string", "index":"not_analyzed"},
-        "ServiceID":       {"type": "string", "index":"not_analyzed"},
-        "DockerID":        {"type": "string", "index":"not_analyzed"},
-        "PrivateIP":       {"type": "string", "index":"not_analyzed"},
-        "Scheduled" :      {"type": "date", "format" : "dateOptionalTime"},
-        "Terminate" :      {"type": "date", "format" : "dateOptionalTime"},
-        "Started" :        {"type": "date", "format" : "dateOptionalTime"},
-        "EndPoints" :      {
-          "properties":    {
-            "Name" :        {"type": "string", "index":"not_analyzed"},
-            "Protocol" :    {"type": "string", "index":"not_analyzed"},
-            "Application" : {"type": "string", "index":"not_analyzed"},
-            "Purpose" :     {"type": "string", "index":"not_analyzed"},
-            "PortNumber" :  {"type": "long",   "index":"not_analyzed"}
-          }
+    "properties": {
+      "Id" :             {"type": "keyword", "index":"true"},
+      "HostID":          {"type": "keyword", "index":"true"},
+      "ServiceID":       {"type": "keyword", "index":"true"},
+      "DockerID":        {"type": "keyword", "index":"true"},
+      "PrivateIP":       {"type": "keyword", "index":"true"},
+      "Scheduled" :      {"type": "date", "format" : "date_optional_time"},
+      "Terminate" :      {"type": "date", "format" : "date_optional_time"},
+      "Started" :        {"type": "date", "format" : "date_optional_time"},
+      "TemplateID":      {"type": "keyword", "index":"true"},
+      "DeployedAt" :     {"type": "date", "format" : "date_optional_time"},
+      "EndPoints" :      {
+        "properties":    {
+          "Name" :        {"type": "keyword", "index":"true"},
+          "Protocol" :    {"type": "keyword", "index":"true"},
+          "Application" : {"type": "keyword", "index":"true"},
+          "Purpose" :     {"type": "keyword", "index":"true"},
+          "PortNumber" :  {"type": "long",   "index":"true"}
         }
       }
-    },
```

Comment:
If the question about comma after bracket, then valid JSON doesn't require it. In general, because of in new ES the mapping types were removed we have significant changes in queries and mapping files 
https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html

Reply:
It was about the "servicedeployment" section being gone. After I wrote this comment I read about the mapping and type changes in elastic.

My next question is if mappings are deprecated why keep them at all?

Reply:
`mapping types` are deprecated but not the `mapping` in general